pkg/reloadwatcher/reloadwatcher.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reloadwatcher
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"istio.io/istio/pkg/log"
+
+	"github.com/howeyc/fsnotify"
+)
+
+var handlers struct {
+	sync.Mutex
+	listeners []chan<- string
+	signals   chan os.Signal
+}
+
+// Notify a channel if a reload event is triggered. A notification is always triggered for SIGUSR1
+func Watch(c chan<- string) {
+	if c == nil {
+		panic("reload: Watch using nil channel")
+	}
+
+	handlers.Lock()
+	defer handlers.Unlock()
+
+	if handlers.listeners == nil {
+		// Watch for SIGUSR1 by default
+		handlers.signals = make(chan os.Signal, 1)
+		signal.Notify(handlers.signals, syscall.SIGUSR1)
+		go func() {
+			for range handlers.signals {
+				Notify("SIGUSR1")
+			}
+		}()
+		handlers.listeners = make([]chan<- string, 0, 10)
+	}
+	handlers.listeners = append(handlers.listeners, c)
+}
+
+// Directly trigger a notification
```

Comment:
You aren't getting scolded by the linters? I think the comments is required to start with // Notify ...

Reply:
Also, this probably does not need to be public.

Reply:
Its triggered in the ctrlz handler so it does need to be public. 
Linter seems happy enough